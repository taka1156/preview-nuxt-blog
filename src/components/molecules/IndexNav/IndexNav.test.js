import { mount } from '@vue/test-utils';
import IndexNav from './IndexNav.vue';

describe('IndexNav', () => {
  const indexNav = propsData => {
    return mount(IndexNav, {
      slots: {
        default: 'TEXT'
      },
      propsData: {
        ...propsData
      }
    });
  };

  it('IndexNav初期値: isOpen', () => {
    const wrapper = indexNav({ isOpen: false });
    // isOpen
    expect(wrapper.vm.$options.props.isOpen.required).toBe(true);
    expect(wrapper.vm.isOpen).toBe(false);
  });

  it('ナビゲーションバーを開く', () => {
    const wrapper = indexNav({ isOpen: true });
    expect(wrapper.vm.isOpen).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('ナビゲーションバーを閉じる', () => {
    const wrapper = indexNav({ isOpen: false });
    expect(wrapper.vm.isOpen).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
