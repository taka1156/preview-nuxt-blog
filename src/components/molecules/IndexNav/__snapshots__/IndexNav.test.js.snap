// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexNav ナビゲーションバーを閉じる 1`] = `
"<div>
  <div class=\\"index-nav\\">
    <div class=\\"nav-icon\\"><button><span class=\\"nav-icon__border nav-icon__top nav-icon__top--close\\"></span> <span class=\\"nav-icon__border nav-icon__middle\\"></span> <span class=\\"nav-icon__border nav-icon__bottom nav-icon__bottom--close\\"></span> <span class=\\"nav-icon__text\\">
      INDEX
    </span></button></div>
  </div>
</div>"
`;

exports[`IndexNav ナビゲーションバーを開く 1`] = `
"<div>
  <div class=\\"index-nav\\">
    <div class=\\"nav-icon\\"><button><span class=\\"nav-icon__border nav-icon__top nav-icon__top--open\\"></span> <span class=\\"nav-icon__border nav-icon__middle nav-icon__middle--fade\\"></span> <span class=\\"nav-icon__border nav-icon__bottom nav-icon__bottom--open\\"></span> <span class=\\"nav-icon__text\\">
      CLOSE
    </span></button></div>
  </div>
</div>"
`;
